tests/shadow/test_shadow_measurement.py

```diff
@@ -0,0 +1,240 @@
+        dev.classical_shadow = super(type(dev), dev).classical_shadow
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        qml.Hadamard(wires=0)
+
+        for target in range(1, wires):
+            qml.CNOT(wires=[0, target])
+
+        return qml.classical_shadow(wires=range(wires), seed_recipes=seed_recipes)
+
+    return circuit
+
+
+def get_x_basis_circuit(wires, shots, interface="autograd", device="default.qubit"):
+    """
+    Return a QNode that prepares the |++..+> state and performs a classical shadow measurement
+    """
+    if device is not None:
+        dev = qml.device(device, wires=wires, shots=shots)
+    else:
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+        # make the device call the superclass method to switch between the general qubit device and device specific implementations (i.e. for default qubit)
+        dev.classical_shadow = super(type(dev), dev).classical_shadow
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        for wire in range(wires):
+            qml.Hadamard(wire)
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+def get_y_basis_circuit(wires, shots, interface="autograd", device="default.qubit"):
+    """
+    Return a QNode that prepares the |+i>|+i>...|+i> state and performs a classical shadow measurement
+    """
+    if device is not None:
+        dev = qml.device(device, wires=wires, shots=shots)
+    else:
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+        # make the device call the superclass method to switch between the general qubit device and device specific implementations (i.e. for default qubit)
+        dev.classical_shadow = super(type(dev), dev).classical_shadow
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        for wire in range(wires):
+            qml.Hadamard(wire)
+            qml.RZ(np.pi / 2, wire)
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+def get_z_basis_circuit(wires, shots, interface="autograd", device="default.qubit"):
+    """
+    Return a QNode that prepares the |00..0> state and performs a classical shadow measurement
+    """
+    if device is not None:
+        dev = qml.device(device, wires=wires, shots=shots)
+    else:
+        dev = qml.device("default.qubit", wires=wires, shots=shots)
+
+        # make the device call the superclass method to switch between the general qubit device and device specific implementations (i.e. for default qubit)
+        dev.classical_shadow = super(type(dev), dev).classical_shadow
+
+    @qml.qnode(dev, interface=interface)
+    def circuit():
+        return qml.classical_shadow(wires=range(wires))
+
+    return circuit
+
+
+class TestShadowMeasurement:
+
+    wires_list = [1, 2, 3, 5, 8]
+    shots_list = [1, 10, 100]
+    seed_recipes_list = [True, False]
```

Comment:
Currently almost all tests are run twice with and without seed recipes fixed. Would it be sufficient to test different seeds in a separate test once and not have it run twice for all tests below? Certainly is more thorough but maybe overkill?

Reply:
After reducing the number of wires (as per your suggestion above), the tests that parametrize over the `seed_recipes` parameter run very quickly now. I'd prefer to leave it as it is to be more thorough.

Reply:
Okay if the wire reduction already does the most part then thats fine 👍